CHANGELOG_UNRELEASED.md

```diff
@@ -52,6 +52,62 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
     `enum_val_bij`, `le_enum_val`, `le_enum_rank_in`, and
     `le_enum_rank`.  They are all accessible via module `Order` if one
     chooses not to import `Order.EnumVal`.
+  + a new module `tagnat` which provides a monotonous bijection
+    between the finite ordered types `{i : 'I_n & 'I_(p_ i)}`
+    (canonically ordered lexicographically), and `'I_(\sum_i p_ i)`,
+    via the functions `sig` and `rank`. We provide direct access to
+    the components of the former type via the functions `sig1`, `sig2`
+    and `Rank`. We provide the following lemmas on these definitions:
+    `card`, `sigK`, `sig_inj`, `rankK`, `rank_inj`, `sigE12`,
+    `rankE`, `sig2K`, `Rank1K`, `Rank2K`, `rank_bij`, `sig_bij `,
+    `rank_bij_on`, `sig_bij_on`, `le_sig`, `le_sig1`, `le_rank`,
+    `le_Rank`, `lt_sig`, `lt_rank`, `lt_Rank`, `eq_Rank`, `rankEsum`,
+    `RankEsum`, `rect`, and `eqRank`.
+
+- in `matrix.v`, seven new definitions:
+  + `mxblock` `mxcol`, `mxrow` and `mxdiag` with notations
+    `\mxblock_(i < m, j < n) B_ i j`, `\mxcol_(i < m) B_ i`,
+    `\mxrow_(j < n) B_ j`, and `\mxdiag_(i < n) B_ i` (and variants
+    without explicit `< n`), to form big matrices described by blocks.
+  + `submxblock`, `submxcol` and `submxrow` to extract blocks from the
```

Comment:
why not submxdiag?

Reply:
This would be redundant with `submxblock A i i`

Reply:
I could add it as a notation, but I'm not convinced it is very useful

Reply:
cf https://github.com/math-comp/math-comp/pull/703/files#diff-f457cb0bf11677ea475ce7d07aa9d8ce79d060ef642432808d92ceb55d6a3fc5R4371-R4373

Reply:
maybe just adding a comment

Reply:
Where and what comment would you think is appropriate?
Something like ``there is no analogous for `mxdiag` because one can simply apply `submxblock A i i` ``?

Reply:
yes in the documention of `matrix.v`.
This makes me realised that there is no doc for `subblock` and the others.

Reply:
See the new revision

Reply:
this should go in the header of `matrix.v` with the documentation of `subblock` and al

Reply:
I added a comment about the absence of `submxdiag` juste below the line you pointed out.
For the other subblocks I am tempted to open an issue instead of fixing it here.

Reply:
I am not sure I undertand. 
You have introduced new constructors `submxblock` ... so you should add their documentation in the header of `matrix`. I think the comment about the absense of `submxdiag` should go there too.

Reply:
Ah you are right, sorry I got confused between the changelog and the header